src/main/java/org/openrewrite/java/testing/junit5/TemporaryFolderToTempDir.java

```diff
@@ -74,206 +78,144 @@ protected TemporaryFolderToTempDirVisitor getVisitor() {
         }
 
         @Override
-        public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
-            J.VariableDeclarations m = super.visitVariableDeclarations(multiVariable, executionContext);
-            if (m.getTypeAsFullyQualified() == null ||
-                    !m.getTypeAsFullyQualified().getFullyQualifiedName()
-                            .equals("org.junit.rules.TemporaryFolder")) {
-                return m;
+        public J visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
+            J.VariableDeclarations mv = (J.VariableDeclarations) super.visitVariableDeclarations(multiVariable, executionContext);
+            if (!isRuleAnnotatedTemporaryFolder(mv)) {
+                return mv;
             }
-            String fieldVars = m.getVariables().stream()
-                    .map(v -> v.withInitializer(null))
+            String fieldVars = mv.getVariables().stream()
+                    .map(fv -> fv.withInitializer(null))
                     .map(J::print).collect(Collectors.joining(","));
-            String modifiers = m.getModifiers().stream().map(it -> it.getType().name().toLowerCase()).collect(Collectors.joining(" "));
-            m = m.withTemplate(
+            String modifiers = mv.getModifiers().stream().map(it -> it.getType().name().toLowerCase()).collect(Collectors.joining(" "));
+            mv = mv.withTemplate(
                     template("@TempDir\n#{} File#{};")
                             .imports("java.io.File", "org.junit.jupiter.api.io.TempDir")
                             .javaParser(TEMPDIR_PARSER::get)
                             .build(),
-                    m.getCoordinates().replace(),
+                    mv.getCoordinates().replace(),
                     modifiers,
                     fieldVars);
-
-            doAfterVisit(new ReplaceTemporaryFolderMethods(m));
-            return m;
+            return mv;
         }
 
-        /**
-         * This visitor replaces methods from TemporaryFolder with JUnit5-compatible alternatives:
-         * <p>
-         * File getRoot()
-         * File newFile()
-         * File newFile(String fileName)
-         * File newFolder()
-         * File newFolder(String... folderNames)
-         * File newFolder(String folder)
-         */
-        private static class ReplaceTemporaryFolderMethods extends JavaVisitor<ExecutionContext> {
-
-            private final J.VariableDeclarations tempDirFields;
-
-            private ReplaceTemporaryFolderMethods(J.VariableDeclarations tempDirFields) {
-                this.tempDirFields = tempDirFields;
-            }
+        private boolean isRuleAnnotatedTemporaryFolder(J.VariableDeclarations vd) {
+            return vd.getTypeAsFullyQualified() != null
+                    && "org.junit.rules.TemporaryFolder".equals(vd.getTypeAsFullyQualified().getFullyQualifiedName())
```

Comment:
This could be simplified with `TypeUtils.isOfType()` or `TypeUtils.isOfClassType()`

Reply:
thanks will make that change